From 6f396335289763e35a0fc5293654713c15ea1df2 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 29 Feb 2016 20:26:39 -0600
Subject: [PATCH] Fix ServerListPingEvent flagging as Async

This event can sometimes fire Async, set the proper boolean

diff --git a/src/main/java/org/bukkit/event/server/ServerEvent.java b/src/main/java/org/bukkit/event/server/ServerEvent.java
index 46b119017..05167fb34 100644
--- a/src/main/java/org/bukkit/event/server/ServerEvent.java
+++ b/src/main/java/org/bukkit/event/server/ServerEvent.java
@@ -1,5 +1,6 @@
 package org.bukkit.event.server;
 
+import org.bukkit.Bukkit;
 import org.bukkit.event.Event;
 
 /**
@@ -8,7 +9,7 @@ import org.bukkit.event.Event;
 public abstract class ServerEvent extends Event {
 
     public ServerEvent() {
-        super();
+        super(!Bukkit.isPrimaryThread()); // Paper
     }
 
     public ServerEvent(boolean isAsync) {
-- 
2.21.0

